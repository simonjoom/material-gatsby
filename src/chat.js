import React, { Component } from "react";
import { graphql, compose } from "react-apollo";
import { Router, Location, globalHistory } from "@reach/router";
import gql from "graphql-tag";
import { AUTH_TOKEN } from "./constants/constants";
import { ApolloProvider } from "react-apollo";
import { ApolloClient } from "apollo-client";
import { createHttpLink } from "apollo-link-http";
import { InMemoryCache } from "apollo-cache-inmemory";
import { setContext } from "apollo-link-context";
import { split } from "apollo-link";
//import { onError } from "apollo-link-error";
//import { ApolloClient, InMemoryCache, HttpLink, split } from 'apollo-client-preset';
import { WebSocketLink } from "apollo-link-ws";
//import { createUploadLink } from "apollo-upload-client";
import { getMainDefinition } from "apollo-utilities";
//import CSSTransition from "react-transition-group/CSSTransition";
import { TransitionGroup, CSSTransition } from "react-transition-group";
import Loading from "./chatcomponents/error/Loading";
import EmailValidated from "./chatcomponents/nav/EmailValidated";
import Header from "./chatcomponents/nav/layout/Header";
import NotFound from "./chatcomponents/error/NotFound";
import { SideBarContext } from "./chatcomponents/SideBarContext";

import UsersPage from "./chatcomponents/user/UsersPage";
import UserPageCreate from "./chatcomponents/user/UserPageCreate";
import UserPage from "./chatcomponents/user/UserPage";
import Logo from "./components/logo";
/*
import CreateCar from "./chatcomponents/car/CreateCar";
import DetailCar from "./chatcomponents/car/DetailCar";
import CarsPage from "./chatcomponents/car/CarsPage";*/
//import Api from "./chatcomponents/api/Api";
import ForgetPassword from "./chatcomponents/user/auth/ForgetPassword";
import Login from "./chatcomponents/user/auth/Login";
import Signup from "./chatcomponents/user/auth/Signup";
import ResetPassword from "./chatcomponents/user/auth/ResetPassword";
import UpdatePassword from "./chatcomponents/user/auth/UpdatePassword";
import ValidateEmail from "./chatcomponents/user/auth/ValidateEmail";
/*import DraftsPage from "./chatcomponents/post/DraftsPage";
import CreatePage from "./chatcomponents/post/CreatePage";
import DetailPage from "./chatcomponents/post/DetailPage";
import FeedPage from "./chatcomponents/post/FeedPage";*/
import ChatsPage from "./chatcomponents/chat/ChatsPage";
import Badge from "./reactLIB/Badge";
import Button from "./reactLIB/Button";

let pathbackend = "https://ns327841.ip-37-187-112.eu/graphql";
let uriwebsocket = "wss://ns327841.ip-37-187-112.eu/subscriptions";
//+process.env.REACT_APP_ENDPOINT;
if (!process.env.NODE_ENV || process.env.NODE_ENV === "development") {
  // dev code
  pathbackend = "http://localhost:4000/graphql";
  uriwebsocket = "ws://localhost:4000/subscriptions";
}

const wsLink = new WebSocketLink({
  uri: uriwebsocket,
  options: {
    reconnect: true,
    connectionParams: () => ({
      authorization: `Bearer ${localStorage.getItem(AUTH_TOKEN)}`
    })
  }
});
const httpLink = new createHttpLink({ uri: pathbackend });

const middlewareAuthLink = setContext((_, { headers }) => {
  console.log("middlewareAuthLink", localStorage.getItem(AUTH_TOKEN));
  // get the authentication token from local storage if it exists
  const token = localStorage.getItem(AUTH_TOKEN);
  // return the headers to the context so httpLink can read them
  return {
    headers: {
      ...headers,
      Authorization: token ? `Bearer ${token}` : ``
    }
  };
});

const httpLinkWithAuthToken = middlewareAuthLink.concat(httpLink);

const link = split(
  ({ query }) => {
    const { kind, operation } = getMainDefinition(query);
    return kind === "OperationDefinition" && operation === "subscription";
  },
  wsLink,
  httpLinkWithAuthToken
);

const client = new ApolloClient({
  link,
  ssrMode: false,
  cache: new InMemoryCache()
});

const USER_QUERY = gql`
  query Me {
    me {
      id
      role
      emailvalidated
    }
  }
`;

const isMobile = () => window && (window.innerWidth < 600 ? true : false);
const Page = ({ page, Teader, startChat }) => {
  // const p = props.page ? 1 : 0;
  // style={{ background: `hsl(${p * 75}, 60%, 60%)` }}
  return (
    <div className="page" id="itc-ctr">
      <div className="itc-messenger itc-messenger-cvs itc-messenger-from-home-screen">
        <div className="itc-messenger-header">
          <div
            aria-label="Expand cvs"
            role="button"
            tabIndex="0"
            className="md-grid md-grid--stacked"
          >
            {Teader()}
            <div
              className="md-cell md-cell--6 md-cell--center md-cell--middle"
              aria-hidden="false"
              style={{ opacity: 1 }}
            >
              <div className="itc-admin-profile-compact-contents">
                <div className="itc-admin-profile-compact-avatar-ctr">
                  <div className="itc-admin-profile-compact-avatar">
                    <div className="itc-avatar">
                      <img
                        src="https://static.intercomassets.com/avatars/1422586/square_128/myAvatar-2-270x250-1509093775.png?1509093775"
                        alt="E-Residency profile"
                      />
                    </div>
                    <div className="itc-active-state" />
                  </div>
                </div>
                <div className="itc-admin-profile-compact-body md-grid md-grid--stacked">
                  <div className="md-cell md-cell--12">Valentina</div>
                  <div className="md-cell md-cell--12">Active</div>
                  <h4 className="md-cell md-cell--12">{startChat}</h4>
                </div>
              </div>
            </div>
          </div>
        </div>

        {page}
      </div>
    </div>
  );
};
class ChatLayoutJSX extends Component {
  state = {
    isSideBarOpen: !isMobile(),
    variant: "permanent",
    isMobile: false
  };
  componentDidMount() {
    console.log("mountLayout");
    console.log("Materialize Ready?", M);
    // this.closeChat = this.closeChat.bind(this);
    var that = this;
    this.manualclose = true;
    var elems = document.querySelectorAll("#myChat .collapsible");
    this.instCollaps = M.Collapsible.init(elems, {
      onOpenStart: () => {
        that.clickonCollaps = true;
        $("body").addClass("stop-scroll");
        that.instanceTap && that.instanceTap.close();
        $("body").addClass("stop-scroll")
      },
      onOpenEnd: () => {
        var objDiv = $("#listChats");
        if (objDiv[0]) {
          var t = objDiv[0].scrollHeight;
          objDiv[0].scrollTop = t;
        }
      }
    });

    var el = document.querySelectorAll(".tap-target");
    var instancestap = M.TapTarget.init(el, {
      onClose: () => {
        if (that.manualclose) {
          var instance = that.instCollaps[0];
          setTimeout(function() {
            instance.open();
          }, 400);
        }
        // console.log("opentrae",instance);
      }
    });
    M.startTextFields();
    const run = localStorage.getItem("Tapinstance");
    if (run !== "run")
      setTimeout(function() {
        // only show the tip after 4sec if the chat Collapsible was never activated on a click
        if (!that.clickonCollaps) {
          that.instanceTap = M.TapTarget.getInstance(instancestap[0].el);
          that.instanceTap.open();
          localStorage.setItem("Tapinstance", "run");
          setTimeout(function() {
            that.manualclose = false;
            that.instanceTap.close();
          }, 5000);
        }
      }, 4800);
  }
  closeChat = () => {
    $("body").removeClass("stop-scroll");
    this.instCollaps[0].close();
    $("body").removeClass("stop-scroll")
  };
  componentWillUnmount() {
    if (this.instCollaps) {
      this.instCollaps.destroy();
    }
    if (this.instance) {
      this.instance.destroy();
    }
  }
  toggleDrawer = () => {
    this.setState({
      isSideBarOpen: !this.state.isSideBarOpen
    });
  };
  /*
*/
  /*
componentDidMount() {
  window.addEventListener("resize", this.resize);
  this.setState({
    isMobile: this.isMobile(),
    variant: this.isMobile() ? "persistent" : "permanent"
    //   isSideBarOpen: this.isMobile() ? false : true
  });
}

resize = () => {
  this.setState({
    isMobile: this.isMobile(),
    variant: this.isMobile() ? "persistent" : "permanent"
    //    isSideBarOpen: this.isMobile() ? false : true
  });
};*/

  render() {
    // const propstoshare = this.props.children ? this.props.children.props : {};
    const { me: Mep, validation } = this.props;
    const authToken = localStorage.getItem(AUTH_TOKEN);
    const startChat = global.tr("Index")("startChat");
    const size = this.state.isSideBarOpen ? "11" : "12";
    const sizet = this.state.isSideBarOpen ? "7" : "8";
    const sizem = this.state.isSideBarOpen ? "3" : "4";
    console.log("Mep",Mep.me)
    let Me = Mep.me;
    return (
      <ul className="collapsible popout">
        <li>
          <Location>
            {({ location }) => {
              const other = {
                startChat,
                Teader: () => (
                  <SideBarContext.Provider
                    value={{
                      toggleDrawer: this.toggleDrawer,
                      state: this.state,
                      Me: !Mep.loading && !Mep.error && Me
                    }}
                  >
                    <Header
                      location={location}
                      onClose={() => this.closeChat()}
                      isSideBarOpen={this.state.isSideBarOpen}
                      isMobile={this.state.isMobile}
                    />
                  </SideBarContext.Provider>
                )
              };
              return (
                <>
                  <div className="collapsible-body itc-messenger-frame itc-messenger-frame-enter-done">
                    <div
                      className="md-grid md-grid--no-spacing"
                      style={{ height: "100%" }}
                    >
                      {/* {!Mep.loading &&
                        !Mep.error &&
                        Me && (
                          <SideBar
                            Me={Me}
                            isSideBarOpen={this.state.isSideBarOpen}
                            isMobile={this.state.isMobile}
                          />
                        )} */}
                      <div
                        className={`md-grid md-grid--no-spacing md-grid--stacked md-cell md-cell--${size} md-cell--${sizet}-tablet md-cell--${sizem}-phone`}
                      >
                        {Mep.loading && <Loading />}

                        <FadeTransitionRouter location={location}>
                          <Page path="/users" page={<UsersPage />} {...other} />
                          <Page
                            path="/user/create"
                            page={<UserPageCreate />}
                            {...other}
                          />
                          <Page
                            path="/user/:id"
                            page={<UserPage path="/user/:id" />}
                            {...other}
                          />
                          <Page path="/chats" page={<ChatsPage />} {...other} />
                          <Page path="/login" page={<Login />} {...other} />
                          <Page path="/signup" page={<Signup />} {...other} />
                          <Page
                            path="/forgetPassword"
                            page={<ForgetPassword />}
                            {...other}
                          />
                          <Page
                            path="/resetPassword"
                            page={<ResetPassword />}
                            {...other}
                          />
                          <Page
                            path="/updatePassword"
                            page={<UpdatePassword />}
                            {...other}
                          />
                          <Page
                            path="/validateEmail"
                            page={<ValidateEmail />}
                            {...other}
                          />
                          <Page
                            path="/"
                            default
                            {...other}
                            page={
                              !authToken ? (
                                <Login path="/" />
                              ) : (
                                <ChatsPage path="/" />
                              )
                            }
                          />
                        </FadeTransitionRouter>
                        {!Mep.loading &&
                          !Mep.error &&
                          validation && (
                            <EmailValidated
                              emailvalidated={Me.emailvalidated}
                            />
                          )}
                      </div>
                    </div>
                  </div>

                  <div
                    className="collapsible-header waves-effect waves-light btn"
                    style={{ display: "flex", alignItems: "center" }}
                  >
                    <Logo id="chat" width={40} height={40} />
                    <p style={{ marginLeft: 20 }}>
                      {global.tr("Index")("Chathello")}?
                    </p>
                    <Badge newIcon>4</Badge>
                  </div>
                  <div className="tap-target bgprimary" data-target="chat">
                    <div className="tap-target-content white">
                      <h5 className="h2 nolineheight">
                        {global.tr("Index")("hello")}
                      </h5>
                      <h6 className="h3 nolineheight">
                        {global.tr("Index")("chatwith")}
                      </h6>
                    </div>
                  </div> 
                </>
              );
            }}
          </Location>
        </li>
      </ul>
    );
  }
}

const FadeTransitionRouter = ({ children, location }) => {
  // const childrenpass = React.cloneElement(children, propstoshare);
  console.log("startFadeTransitionRouter", location);
  return (
    <TransitionGroup className="transition-group">
      <CSSTransition key={location.key} classNames="fade" timeout={400}>
        <Router location={location} className="router">
          {children}
        </Router>
      </CSSTransition>
    </TransitionGroup>
  );
};

ChatLayoutJSX.defaultProps = {
  validation: false,
  me: {}
};

const ChatLayout = compose(
  graphql(USER_QUERY, {
    name: "me",
    options: props => {
      return {
        fetchPolicy: "cache-and-network",
        ssr: false
      };
    }
  })
)(ChatLayoutJSX);

const Chat = () => {
  return (
    <ApolloProvider client={client}>
      <ChatLayout />
    </ApolloProvider>
  );
};
export default Chat;
