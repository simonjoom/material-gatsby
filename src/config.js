const config = {
  locales: ["fr", "en", "pt", "ru", "uk", "ch"],
  defaultLocale: "en",
  router: {
    "/": {
      fr: "/fr/",
      en: "/",
      ru: "/ru/",
      uk: "/uk/",
      pt: "/pt/",
      ch: "/cn/"
    },
    "/about/": {
      fr: "/fr/L_ecole_de_ski/",
      en: "/About_Skiscool/",
      ru: "/ru/Около_Skiscool/",
      uk: "/uk/про_Skiscool/",
      pt: "/pt/About_Skiscool/",
      ch: "/cn/About_Skiscool/"
    },
    "/404/": {
      fr: "/fr/404/",
      en: "/404/",
      ru: "/ru/404/",
      uk: "/uk/404/",
      pt: "/pt/404/",
      ch: "/cn/404/"
    },
    "/blog/": {
      fr: "/fr/ecole-de-ski-courchevel-meribel/",
      en: "/blog/",
      ru: "/ru/лыжная-школа-Куршевель/",
      uk: "/uk/Лижна-школа-Куршевель/",
      pt: "/pt/blog/",
      ch: "/cn/blog/"
    },
    "/instructor/": {
      fr: "/fr/Moniteurs_skischool/",
      en: "/Instructors_skischool/",
      ru: "/ru/Инструкторы_skischool/",
      uk: "/uk/інструкторів_skischool/",
      pt: "/pt/instructors_skischool/",
      ch: "/cn/Instructors_skischool/"
    },
    "/hotel/": {
      fr: "/fr/Hotels/",
      en: "/Hotels/",
      ru: "/ru/Гостиница/",
      uk: "/uk/отель/",
      pt: "/pt/Hoteis/",
      ch: "/cn/Hotels/"
    },
    "/hotel": {
      fr: "/fr/Hotels/",
      en: "/hotels/",
      ru: "/ru/Гостиница/",
      uk: "/uk/отель/",
      pt: "/pt/Hoteis/",
      ch: "/cn/Hotels/"
    },
    "/instructor": {
      fr: "/fr/Moniteurs_skischool/",
      en: "/Instructors_skischool/",
      ru: "/ru/Инструкторы_skischool/",
      uk: "/uk/інструкторів_skischool/",
      pt: "/pt/instructors_skischool/",
      ch: "/cn/Instructors_skischool/"
    },
    "/concept/": {
      fr: "/fr/concept",
      en: "/concept/",
      ru: "/ru/concept/",
      uk: "/uk/concept/",
      pt: "/pt/concept/",
      ch: "/cn/concept/"
    },
    "/contact/": {
      fr: "/fr/contact",
      en: "/contact/",
      ru: "/ru/контакт/",
      uk: "/uk/дотик/",
      pt: "/pt/contato/",
      ch: "/cn/contact/"
    },
    "/skipass/": {
      fr: "/fr/Articles/Forfait-de-ski/",
      en: "/Articles/SkiPass/",
      ru: "/ru/Articles/Ски-пасс/",
      uk: "/uk/Articles/Ски-пас/",
      pt: "/pt/Articles/SkiPass/",
      ch: "/cn/Articles/SkiPass/"
    },
    "/meribel/": {
      fr: "/fr/Articles/Meribel/",
      en: "/Articles/Meribel/",
      ru: "/ru/Articles/Meribel/",
      uk: "/uk/Articles/Meribel/",
      pt: "/pt/Articles/Meribel/",
      ch: "/cn/Articles/Meribel/"
    },
    "/menuires/": {
      fr: "/fr/Articles/Menuires/",
      en: "/Articles/Menuires/",
      ru: "/ru/Articles/Menuires/",
      uk: "/uk/Articles/Menuires/",
      pt: "/pt/Articles/Menuires/",
      ch: "/cn/Articles/Menuires/"
    },
    "/courchevel/": {
      fr: "/fr/Articles/Courchevel/",
      en: "/Articles/Courchevel/",
      ru: "/ru/Articles/Courchevel/",
      uk: "/uk/Articles/Courchevel/",
      pt: "/pt/Articles/Courchevel/",
      ch: "/cn/Articles/Courchevel/"
    },
    "/valthorens/": {
      fr: "/fr/Articles/Valthorens/",
      en: "/Articles/Valthorens/",
      ru: "/ru/Articles/Valthorens/",
      uk: "/uk/Articles/Valthorens/",
      pt: "/pt/Articles/Valthorens/",
      ch: "/cn/Articles/Valthorens/"
    },
    "/latania/": {
      fr: "/fr/Articles/Tania/",
      en: "/Articles/Tania/",
      ru: "/ru/Articles/Tania/",
      uk: "/uk/Articles/Tania/",
      pt: "/pt/Articles/Tania/",
      ch: "/cn/Articles/Tania/"
    }
  }
};

module.exports = config;
//export default config;
/*
    "/jumpsuit": {
      fr: "/jumpsuit/fr/",
      en: "/jumpsuit/",
      ru: "/ru/jumpsuit/",
      uk: "/uk/jumpsuit/",
      pt: "/pt/jumpsuit/",
      ch: "/cn/jumpsuit/"
    },*/