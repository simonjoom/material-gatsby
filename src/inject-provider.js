import React from "react";
import i18n from "i18next";
import { ThemeContext } from "./withContext";
var canUseDOM = !!(
  typeof window !== "undefined" &&
  window.document &&
  window.document.createElement
);

const filetranslate = lng => import(`./layouts/translate_${lng}`);
const ZeptoAsync = () =>
  import(/* webpackChunkName: "zepto" */ "./components/zepto");
const Statics = () =>
  import(/* webpackChunkName: "statics" */ "./layouts/statics");

global.menuList = { en: [], ru: [], pt: [], uk: [], ch: [], fr: [] };
global.filesQuery = [];
const preferDefault = m => (m && m.default) || m;
let Layout;
try {
  Layout = preferDefault(require(GATSBY_LAYOUT_COMPONENT_PATH));
} catch (e) {
  if (e.toString().indexOf(`Error: Cannot find module`) !== -1) {
    throw new Error(
      `Couldn't find layout component at "${GATSBY_LAYOUT_COMPONENT_PATH}.\n\n` +
        `Please create layout component in that location or specify path to layout component in gatsby-config.js`
    );
  } else {
    throw e;
  }
}

global.Zepto = undefined;
const getLanguage = () =>
  i18n.language ||
  (typeof window !== "undefined" && window.localStorage.i18nextLng);

export const wrapPageElement = async ({ element, props }) => {
  const { lng } = props.pageContext;
  let Red, Red2;
  if (global.menuList[lng].length == 0 || getLanguage() !== lng) {
    const Obj = await filetranslate(lng);
    Red = Obj.default;
  } else {
    Red = "div";
  }
  if (canUseDOM && !global.Zepto) {
    const Zep = await ZeptoAsync();
    if (/comp|inter|loaded/.test(document.readyState)) {
      Waves.displayEffect();
    } else {
      document.addEventListener(
        "DOMContentLoaded",
        function() {
          Waves.displayEffect();
        },
        false
      );
    }
    global.Zepto = Zep.default;
  }

  if (global.filesQuery.length == 0) {
    const Obj = await Statics();
    Red2 = Obj.default;
  } else {
    Red2 = "div";
  }
  console.log("Red2", Red2, lng);
  const { slug, slugbase, route, carousel } = props.pageContext;
  const namespace = slugbase === "/" ? "Index" : "Post";
  const ismain = slugbase === "/";
  console.log("runMainNavLayout", props.location);
  return (
    <>
      <Red />
      <Red2 lng={lng} />
      <ThemeContext.Provider
        value={{
          translate: namespace => i18n.getFixedT(null, [namespace, "common"])
        }}
      >
        <Layout route={route} lng={lng}>
          {element}
        </Layout>
      </ThemeContext.Provider>
    </>
  );
};

// eslint-disable-next-line react/prop-types,react/display-name
/*export default ({ element }) => (
  <div>{element}</div>
)*/

//<ApolloProvider client={apolloClient}>{element}</ApolloProvider>
