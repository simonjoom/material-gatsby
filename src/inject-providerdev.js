import React from "react";
import i18n from "i18next";
import { AppRegistry } from "react-native-web";
import { ThemeContext } from "./withContext";  
global.Zepto = undefined;
const getLanguage = () =>
  i18n.language ||
  (typeof window !== "undefined" && window.localStorage.i18nextLng);

var canUseDOM = !!(
  typeof window !== "undefined" &&
  window.document &&
  window.document.createElement
);

global.isSSR=!canUseDOM

const filetranslate = lng => import(`./layouts/translate_${lng}`);
const ZeptoAsync = () =>
  import(/* webpackChunkName: "zepto" */ "./components/zepto");
const Statics = () =>
  import(/* webpackChunkName: "statics" */ "./layouts/statics");
  global.locale= { en: [], ru: [], pt: [], uk: [], ch: [], fr: [] };
global.menuList = { en: [], ru: [], pt: [], uk: [], ch: [], fr: [] };
global.filesQuery = [];
const preferDefault = m => (m && m.default) || m;
let Layout;
try {
  Layout = preferDefault(require(GATSBY_LAYOUT_COMPONENT_PATH));
} catch (e) {
  if (e.toString().indexOf(`Error: Cannot find module`) !== -1) {
    throw new Error(
      `Couldn't find layout component at "${GATSBY_LAYOUT_COMPONENT_PATH}.\n\n` +
        `Please create layout component in that location or specify path to layout component in gatsby-config.js`
    );
  } else {
    throw e;
  }
}

export const replaceHydrateFunction = async () => {
  if (!global.Zepto) {
    const Zep = await ZeptoAsync();
    global.Zepto = Zep.default;
  }
  return (element, container, callback) => {
    class App extends React.Component {
      render() {
        return <div id="App">{element}</div>;
      }
    }

    if (/comp|inter|loaded/.test(document.readyState)) {
      Waves.displayEffect();
    } else {
      document.addEventListener(
        "DOMContentLoaded",
        function() {
          Waves.displayEffect();
        },
        false
      );
    }
    
    AppRegistry.registerComponent("App", () => App);
    AppRegistry.runApplication("App", {
      initialProps: {},
      rootTag: container,
      callback
    });
  };
};

export const wrapPageElement = async ({ element, props }) => {
  const { lng } = props.pageContext;
  let Red, Red2;
  if (global.menuList[lng].length == 0) {
    try {
      const Obj = await filetranslate(lng);
      Red = Obj.default;
    } catch (e) { 
      Red = "div";
    }
  } else {
    Red = "div";
  } 

  if (global.filesQuery.length == 0) {
    const Obj = await Statics();
    Red2 = Obj.default;
  } else {
    Red2 = "div";
  }  
  // console.log("Red2", Red2, lng);
  const { slug, slugbase, route, carousel } = props.pageContext;
  const namespace = slugbase === "/" ? "Index" : "Post";
  const ismain = slugbase === "/";
  console.log("runMainNavLayout");
  return (
    <>
      <Red />
      <Red2 lng={lng} />
      <Layout lng={lng}>{element}</Layout>
    </>
  );
};

export const wrapRootElement = ({ element }) => { 
  return (
    <ThemeContext.Provider
      value={{
        translate: namespace => i18n.getFixedT(null, [namespace, "common"])
      }}
    >
      {element}
    </ThemeContext.Provider>
  );
};