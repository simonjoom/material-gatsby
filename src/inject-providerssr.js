import React from "react";
import i18n from "i18next";
import { ThemeContext } from "./withContext";
import trfr from "./layouts/translate_fr";
import tren from "./layouts/translate_en";
import trpt from "./layouts/translate_pt";
import trru from "./layouts/translate_ru";
import truk from "./layouts/translate_uk";
import trch from "./layouts/translate_ch";
import statics from "./layouts/statics";

global.Zepto = undefined;
const getLanguage = () => i18n.language;

global.isSSR = true;

global.menuList = { en: [], ru: [], pt: [], uk: [], ch: [], fr: [] };
global.filesQuery = [];
const preferDefault = m => (m && m.default) || m;
let Layout;
try {
  Layout = preferDefault(require(GATSBY_LAYOUT_COMPONENT_PATH));
} catch (e) {
  if (e.toString().indexOf(`Error: Cannot find module`) !== -1) {
    throw new Error(
      `Couldn't find layout component at "${GATSBY_LAYOUT_COMPONENT_PATH}.\n\n` +
        `Please create layout component in that location or specify path to layout component in gatsby-config.js`
    );
  } else {
    throw e;
  }
}

export const wrapPageElement = async ({ element, props }) => {
  const { lng } = props.pageContext;
  let Red, Red2;
  if (global.menuList[lng].length == 0 || getLanguage() !== lng) {
    Red =
      lng == "fr"
        ? trfr
        : lng == "pt"
          ? trpt
          : lng == "uk"
            ? truk
            : lng == "ru"
              ? trru
              : lng == "ch"
                ? trch
                : tren;
  } else {
    Red = "div";
  }

  if (global.filesQuery.length == 0) {
    Red2 = statics;
  } else {
    Red2 = "div";
  }
  // console.log("Red2", Red2, lng);
  const { slug, slugbase, route, carousel } = props.pageContext;
  const namespace = slugbase === "/" ? "Index" : "Post";
  const ismain = slugbase === "/";
  console.log("runMainNavLayout", lng);
  return (
    <>
      <Red />
      <Red2 lng={lng} />
      <ThemeContext.Provider
        value={{
          translate: namespace => i18n.getFixedT(null, [namespace, "common"])
        }}
      >
        <Layout route={route} lng={lng}>
          {element}
        </Layout>
      </ThemeContext.Provider>
    </>
  );
};
