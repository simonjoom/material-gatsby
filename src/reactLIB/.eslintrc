# JavaScript Standard Style Guide: http://standardjs.com
# https://github.com/feross/eslint-config-standard-react
extends:
  - standard-react
  - prettier

globals:
  Materialize: true
  React: true
  ReactDOM: true
  M: true
  document: true
  global: true
  window: true

plugins: ['prettier']

env:
  jquery: true
  jest: true

rules:
  prettier/prettier: [2]
  react/jsx-no-bind: [0]
  semi: [2, 'always']
  react/no-unused-prop-types: [0]
  react/jsx-curly-spacing: [0]
  react/jsx-equals-spacing: [0]
  react/jsx-indent: [0]
  react/jsx-indent-props: [0]
  react/jsx-space-before-closing: [0]
  standard/array-bracket-even-spacing: [0]
  standard/computed-property-even-spacing: [0]
  standard/object-curly-even-spacing: [0]
  no-undef: [2]
