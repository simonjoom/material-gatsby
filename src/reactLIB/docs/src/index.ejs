<!DOCTYPE html>
<html>
  <head>
    <title>React Materialize</title>
    <meta http-equiv="Content-type" content="text/html; charset=utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link rel="shortcut icon" href="img/react-materialize-logo.png">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
    <link href="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0-rc.2/css/materialize.css" rel="stylesheet">
  </head>
  <body>
    <div id="root"></div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0-rc.2/js/materialize.js"></script>
  </body>
</html>
