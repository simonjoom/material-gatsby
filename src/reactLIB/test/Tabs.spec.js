import React from 'react';
import { shallow, mount } from 'enzyme';
import Tabs from '../src/Tabs';
import Tab from '../src/Tab';
import mocker from './helper/new-mocker';
import * as idgenHolder from '../src/idgen';

describe('Tabs', () => {
  let wrapper;
  const tabsInitMock = jest.fn();
  const tabInstanceDestroyMock = jest.fn();
  const tabsMock = {
    init: (el, options) => {
      tabsInitMock(options);
      return {
        destroy: tabInstanceDestroyMock
      };
    }
  };
  const restore = mocker('Tabs', tabsMock);
  const idgenSpy = jest.spyOn(idgenHolder, 'default');

  const options = {
    duration: 400,
    onShow: jest.fn(),
    swipeable: true,
    responsiveThreshold: 2
  };

  wrapper = shallow(
    <Tabs>
      <Tab title="one">One</Tab>
      <Tab title="Two">Two</Tab>
    </Tabs>
  );

  afterAll(() => {
    restore();
    idgenSpy.mockRestore();
  });

  beforeEach(() => {
    tabsInitMock.mockClear();
  });

  test('should create list of Tab itemt', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('with options', () => {
    beforeEach(() => {
      wrapper = mount(
        <Tabs tabOptions={options}>
          <Tab title="one">One</Tab>
          <Tab title="Two">Two</Tab>
        </Tabs>
      );
    });

    test('initializes Tabs with options', () => {
      expect(tabsInitMock).toHaveBeenCalledWith(options);
    });
  });

  describe('when updated', () => {
    beforeEach(() => {
      idgenSpy.mockClear();
      wrapper = mount(
        <Tabs tabOptions={options}>
          <Tab title="one">One</Tab>
          <Tab title="Two">Two</Tab>
        </Tabs>
      );
    });

    test('should re-initialize with options', () => {
      expect(tabsInitMock).toHaveBeenCalledWith(options);
      tabsInitMock.mockClear();
      tabInstanceDestroyMock.mockClear();
      wrapper.setProps({ className: 'test' });
      expect(tabInstanceDestroyMock).toHaveBeenCalled();
      expect(tabsInitMock).toHaveBeenCalledWith(options);
    });

    test('idgen is not called after update', () => {
      expect(idgenSpy).toHaveBeenCalledTimes(1);
      wrapper.setProps({ className: 'test' });
      expect(idgenSpy).toHaveBeenCalledTimes(1);
    });
  });

  describe('when unmounted', () => {
    beforeEach(() => {
      tabInstanceDestroyMock.mockClear();
      wrapper.unmount();
    });

    test('should be destroyed', () => {
      expect(tabInstanceDestroyMock).toHaveBeenCalled();
    });
  });
});
