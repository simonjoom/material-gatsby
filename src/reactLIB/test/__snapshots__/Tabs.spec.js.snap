// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should create list of Tab itemt 1`] = `
<Row>
  <Col
    s={12}
  >
    <ul
      className="tabs"
    >
      <li
        className="tab col"
        key="00/.0"
      >
        <a
          className=""
          href="#tab_00"
          onClick={[Function]}
        >
          one
        </a>
      </li>
      <li
        className="tab col"
        key="01/.1"
      >
        <a
          className=""
          href="#tab_01"
          onClick={[Function]}
        >
          Two
        </a>
      </li>
    </ul>
  </Col>
  <Col
    id="tab_00"
    key="tab00/.0"
    s={12}
    style={
      Object {
        "display": "none",
      }
    }
  >
    One
  </Col>
  <Col
    id="tab_01"
    key="tab01/.1"
    s={12}
    style={
      Object {
        "display": "none",
      }
    }
  >
    Two
  </Col>
</Row>
`;
